### PR TITLE
Hide all dependencies on 'proc_macro' under "proc-macro" feature flag

### PR DESCRIPTION
**Pull request by ffuugoo**
Quick fix for #36.

### CODE REVIEW

**Comment by mystor** [comment]
I would want to copy the interface for controlling this which syn and quote use. They define a default-enabled `proc_macro` feature which enables or disables using that `extern crate` decl in both the current crate and dependencies. https://github.com/dtolnay/syn/blob/fa068490fdce0279d37f1d1d8364e68effcb372d/Cargo.toml#L34

In addition, for wasm32-unknown-unknown and friends, they include extra checks to disable the `extern crate` and related methods directly, such as here: https://github.com/dtolnay/syn/blob/fa068490fdce0279d37f1d1d8364e68effcb372d/src/lib.rs#L279-L283

**Comment by ffuugoo** [comment]
@mystor Updated.

File: src/lib.rs [hunk 1]
```diff
@@ -42,6 +42,7 @@
 //!  * Test Case
 //!  */
 //! fn main() {
+//! #   #[cfg(feature = "macro")]
```
Comment (mystor) [inline src/lib.rs]: Please update this feature check

File: src/lib.rs [hunk 2]
```diff
@@ -112,6 +113,7 @@
...
```
Comment (mystor) [inline src/lib.rs]: Please update this feature check

File: src/lib.rs [hunk 3]
```diff
@@ -2311,6 +2321,10 @@ fn trim_start_matches(s: &str, c: char) -> &str {
 
 /// Helper trait describing values which may be returned by macro implementation
 /// methods used by this crate's macros.
+#[cfg(all(
+    not(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "wasi"))),
+    feature = "proc-macro"
+))]
 pub trait MacroResult {
```
Comment (mystor) [inline src/lib.rs]: Let's keep this trait available with the feature disabled, and provide a default implementation of `fn into_stream(self)` which is provided if the feature is enabled. That way we can keep the `test_derive!` macro available on wasm.

File: src/lib.rs (hunk 1 above)
Comment (ffuugoo) [reply src/lib.rs]: Updated.

File: src/lib.rs (hunk 2 above)
Comment (ffuugoo) [reply src/lib.rs]: Updated.

File: src/lib.rs (hunk 3 above)
Comment (ffuugoo) [reply src/lib.rs]: I don't get how `fn into_stream(self)` default implementation should look like. Can you elaborate here?

Comment (mystor) [reply src/lib.rs]: I'll do the change in a follow-up commit, it's not a super substantial one :-)

**Pull request merged**